daemon/cmd/state.go

```diff
@@ -530,6 +525,11 @@ func (d *Daemon) initRestore(restoredEndpoints *endpointRestoreState, endpointsR
 				syncServices(false /* all services */)
 			}
 		}()
+
+		// When we regenerate restored endpoints, it is guaranteed that we have
+		// received the full list of policies present at the time the daemon
+		// is bootstrapped.
+		d.regenerateRestoredEndpoints(restoredEndpoints, endpointsRegenerator)
```

Comment:
Nit. This change looks a no-op, and could be reverted.

Reply:
Not a no-op. The ordering matters.

Reply:
FWIW, I originally also thought this was a no-op. However, I think `regenerateRestoredEndpoints` now waits for the spawned go-routine give a "sync" signal. But it's not really clear to me either what signal it is.

Reply:
Chatted offline, I missed that `regenerateRestoredEndpoints` could block if ipsec is enabled, and so we want to ensure that the goroutine is always started before to prevent a deadlock.